File: Linked list/328. Odd Even Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *oddEvenList(ListNode *head)
    {
        if (head == NULL || head->next == NULL)
            return head;
        ListNode *head1 = head;
        ListNode *head2 = head->next;
        ListNode *temp, *prev;
        temp = head2;
        prev = head1;
        while (temp)
        {
            prev->next = temp->next;
            prev = temp;
            temp = temp->next;
        }
        while (head1->next)
            head1 = head1->next;
        head1->next = head2;
        return head;
    }
};

File: Linked list/328. Odd Even Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        ListNode odd=new ListNode(0);
        ListNode temp_odd=odd;
        ListNode even=new ListNode(0);
        ListNode temp_even=even;
        int c=1;
        while(head!=null)
        {
            if(c%2==1)
            {
                odd.next=head;
                odd=odd.next;
                c++;
            }
            else
            {
                even.next=head;
                even=even.next;
                c++;
            }
            head=head.next;
        }
        odd.next=temp_even.next;
        even.next=null;
        return temp_odd.next;
        
    }
}